colcon_ros/task/catkin/build.py

```diff
@@ -98,12 +99,20 @@ def add_arguments(self, *, parser):  # noqa: D102
                 try:
                     # try to set CATKIN_ENV_HOOK_WORKSPACE explicitly before
                     # sourcing these hooks
-                    additional_hooks.append(
-                        shell_extension.create_hook_set_value(
-                            'catkin_env_hook_workspace',
-                            Path(args.install_base), self.context.pkg.name,
-                            'CATKIN_ENV_HOOK_WORKSPACE',
-                            '$COLCON_CURRENT_PREFIX'))
+                    if sys.platform == 'win32':
+                        additional_hooks.append(
+                            shell_extension.create_hook_set_value(
+                                'catkin_env_hook_workspace',
+                                Path(args.install_base), self.context.pkg.name,
+                                'CATKIN_ENV_HOOK_WORKSPACE',
+                                '%COLCON_CURRENT_PREFIX%'))
```

Comment:
This will only work for `bat` but I would assume not for `PowerShell`. So this might need a `shell_extension` specific expansion.

Reply:
Do you mean something like this? https://github.com/colcon/colcon-core/pull/294